File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/repository/business/ContactPersonRepository.java
package org.d3javu.backend.repository.business;

import org.d3javu.backend.model.business.client.ContactPerson;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.transaction.annotation.Transactional;

public interface ContactPersonRepository extends JpaRepository<ContactPerson, Long> {
    @Query(value = "insert into contact_person(surname, name, patronymic, phone_number, email, business_client_id) " +
            "values(:surname, :name, :patronymic, :phoneNumber, :email, :businessClientId) returning id", nativeQuery = true)
    Long create(Long businessClientId, String surname, String name, String patronymic, String phoneNumber, String email);


    @Transactional
    @Query(value = "update contact_person set " +
            "surname= :surname, name= :name, patronymic= :patronymic, phone_number= :phoneNumber, email= :email " +
            "where id= :id", nativeQuery = true)
    @Modifying
    void update(Long businessClientId, Long id, String surname, String name, String patronymic, String phoneNumber, String email);

    @Transactional
    @Query(value = "delete from contact_person where id=:personId", nativeQuery = true)
    @Modifying
    void delete(Long businessClientId, Long personId);
}

File: Apps/BankCore/backend/src/main/scala/repository/business/contactperson/ContactPersonRepository.scala
package org.d3javu
package repository.business.contactperson

import org.d3javu.domain.business.ContactPerson.{
  Email, Id, Name, Patronymic, PhoneNumber, Surname,
}

import domain.business.Client

trait ContactPersonRepository[F[_]] {
  def create(
      businessClientId: Client.Id,
      surname: Surname,
      name: Name,
      patronymic: Patronymic,
      phoneNumber: PhoneNumber,
      email: Email,
  ): F[Id]

  def update(
      businessClientId: Client.Id,
      id: Id,
      surname: Surname,
      name: Name,
      patronymic: Patronymic,
      phoneNumber: PhoneNumber,
      email: Email,
  ): F[Unit]

  def delete(businessClientId: Client.Id, personId: Id): F[Unit]
}
